src/ResourceManager/Sql/Commands.Sql/Database/Cmdlet/SetAzureSqlDatabase.cs

```diff
@@ -113,7 +113,7 @@ protected override IEnumerable<AzureSqlDatabaseModel> ApplyUserInputToModel(IEnu
                 Edition = Edition,
                 MaxSizeBytes = MaxSizeBytes,
                 RequestedServiceObjectiveName = RequestedServiceObjectiveName,
-                Tags = TagsConversionHelper.CreateTagDictionary(Tags, validate: true),
+                Tags = this.MyInvocation.BoundParameters.ContainsKey("Tags") ? TagsConversionHelper.CreateTagDictionary(Tags, validate: true) : model.FirstOrDefault().Tags,
```

Comment:
If you know that the collection is non-empty, then using `FirstOrDefault()` is just confusing to the reader - use `First()` instead.

The exact same code in 3 places is enough justification to extract method. It can be in utility class or base class, your choice. The fact that the code is complex enough to have a bug once (this time) means that it's complex enough to have a bug again, and when that happens the fix should only be done in one place.